Source/MQTTnet/Client/Connecting/MqttClientAuthenticateResult.cs

```diff
@@ -7,42 +7,110 @@ namespace MQTTnet.Client.Connecting
     // TODO: Consider renaming this to _MqttClientConnectResult_
     public class MqttClientAuthenticateResult
     {
+        /// <summary>
+        /// Gets or sets the result code.
+        /// Hint: MQTT 5 feature only.
+        /// </summary>
         public MqttClientConnectResultCode ResultCode { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether the session is present or not.
+        /// </summary>
         public bool IsSessionPresent { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether wildcard subscriptions are available.
+        /// </summary>
         public bool? WildcardSubscriptionAvailable { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether retained messages are available.
+        /// </summary>
         public bool? RetainAvailable { get; set; }
 
+        /// <summary>
+        /// Gets or sets the assigned client identifier.
+        /// </summary>
         public string AssignedClientIdentifier { get; set; }
 
+        /// <summary>
+        /// Gets or sets the authentication method.
+        /// Hint: MQTT 5 feature only.
+        /// </summary>
         public string AuthenticationMethod { get; set; }
 
+        /// <summary>
+        /// Gets or sets the authentication data.
+        /// Hint: MQTT 5 feature only.
+        /// </summary>
         public byte[] AuthenticationData { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum packet size.
+        /// </summary>
         public uint? MaximumPacketSize { get; set; }
 
+        /// <summary>
+        /// Gets or sets the reason string.
+        /// Hint: MQTT 5 feature only.
+        /// </summary>
         public string ReasonString { get; set; }
 
+        /// <summary>
+        /// Gets or sets the receive maximum.
+        /// </summary>
         public ushort? ReceiveMaximum { get; set; }
         
+        /// <summary>
+        /// Gets or sets the maximum quality od service level.
+        /// </summary>
         public MqttQualityOfServiceLevel MaximumQoS { get; set; }
 
+        /// <summary>
+        /// Gets or sets the response information.
+        /// Hint: MQTT 5 feature only.
+        /// </summary>
         public string ResponseInformation { get; set; }
 
+        /// <summary>
+        /// Gets or sets the topic alias maximum.
+        /// </summary>
         public ushort? TopicAliasMaximum { get; set; }
 
+        /// <summary>
+        /// Gets or sets the server reference.
+        /// </summary>
         public string ServerReference { get; set; }
 
+        /// <summary>
+        /// Gets or sets the server keep alive.
+        /// </summary>
         public ushort? ServerKeepAlive { get; set; }
 
+        /// <summary>
+        /// Gets or sets the session expiry interval.
+        /// </summary>
         public uint? SessionExpiryInterval { get; set; }
```

Comment:
This is also a not valuable comment but maybe adding what happens when the interval is set to null might be interesting or the unit which is seconds here.